File: app/src/main/java/au/id/blackwell/kurt/lantv/TvChannel.java
package au.id.blackwell.kurt.lantv;

import android.net.Uri;

class TvChannel {
    private String mId;
    private String mTitle;
    private String mPlayer;
    private Uri mMediaResolveUri;

    public TvChannel(String id, String title, String player, String mediaResolveUri) {
        this(id, title, player, Uri.parse(mediaResolveUri));
    }

    public TvChannel(String id, String title, String player, Uri mediaResolveUri) {
        mId = id;
        mTitle = title;
        mPlayer = player;
        mMediaResolveUri = mediaResolveUri;
    }

    public String getId() {
        return mId;
    }

    public String getTitle() {
        return mTitle;
    }

    public String getPlayerType() { return mPlayer; }

    public Uri getMediaResolveUri() {
        return mMediaResolveUri;
    }

    public boolean isPlayable() {
        return mPlayer != null && mMediaResolveUri != null;
    }
}

File: app/src/main/java/au/id/blackwell/kurt/lantv/TvChannel.kt
package au.id.blackwell.kurt.lantv

import android.net.Uri

internal data class TvChannel(val id: String, val title: String, val playerType: String?, val mediaResolveUri: Uri) {

    val isPlayable: Boolean
        get() = playerType != null

    constructor(id: String, title: String, player: String?, mediaResolveUri: String) : this(id, title, player, Uri.parse(mediaResolveUri)) {}
}
